src/mon/OSDMonitor.cc

```diff
@@ -4819,8 +4823,12 @@ bool OSDMonitor::prepare_unset_flag(MonOpRequestRef op, int flag)
   ostringstream ss;
   if (pending_inc.new_flags < 0)
     pending_inc.new_flags = osdmap.get_flags();
-  pending_inc.new_flags &= ~flag;
-  ss << "unset " << OSDMap::get_flag_string(flag);
+  if (!osdmap.test_flag(flag)) {
+    ss << OSDMap::get_flag_string(flag) << " flag is not set";
```

Comment:
ditto. and maybe you can move the wait_for* and return true into the `else` block also, personally, i think it's more symmetry, and hence looks better =)